woocommerce/woocommerce-ios
# Product Preview: Add WebKitViewController to handle authenticated sessions

Closes https://github.com/woocommerce/woocommerce-ios/issues/7911, https://github.com/woocommerce/woocommerce-ios/issues/7946.

## Description

This PR adds `WebKitViewController` that uses access token for current user to authenticate them in the context of `WKWebView`.
Most of the code copied from [WordPress-iOS](https://github.com/wordpress-mobile/WordPress-iOS/).

## Testing

Test on the device, as iOS Simulator has [a bug around cookies management](https://github.com/woocommerce/woocommerce-ios/pull/7974#issuecomment-1300776912).

1. Switch to WP.com store.
2. Open products tab.
3. Start the flow to create new product ("+" button in navbar on product list)
4. Edit the title (or anything else) to have non-blank product.
5. Tap the "Preview" button appeared in navbar.
6. Confirm that webpage for draft product loads correctly.
7. Open user profile and tap "Log out". Close the webview.
8. Tap the "Preview" button in navbar again.
9. Confirm that webpage for draft product again loads correctly, authenticated for current user.

## Video

https://user-images.githubusercontent.com/3132438/199501283-90940aa9-0476-400b-90f7-2b63dafbea91.mp4

---
